### ISSUE TITLE
WAN 2.1 I2V Lora not loading (I think?)

### ISSUE THREAD

**Issue by oliverban** [tdrussell/diffusion-pipe#145]
Trained a Lora on Wan 2.1 I2V 14B. Training went fine and loss was good.

Loading the Lora into an updated ComfyUI with the Load Lora Model Only node I get:

......
lora key not loaded: diffusion_model.blocks.9.self_attn.k.lora_B.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.o.lora_A.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.o.lora_B.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.q.lora_A.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.q.lora_B.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.v.lora_A.weight
lora key not loaded: diffusion_model.blocks.9.self_attn.v.lora_B.weight
.....

This list goes on and on. So the lora is not loaded. Any idea why this might be?

Cheers!

**Comment by Vova-B** [comment]
@oliverban Can you tell me what you used to prepare the dataset and share your config for training the lore?

**Comment by abc8350712** [comment]
Same proble.I don't know how to use lora trained by i2v.

**Comment by tdrussell** [comment]
How are you trying to use the lora, native ComfyUI nodes or Kijai's wrapper extension? Have you tried existing loras (you can find several on civit)?

My own trained i2v and t2v loras have exactly the same keys inside them, and they load fine on the fully updated ComfyUI native nodes.

**Comment by oliverban** [comment]
> [@oliverban](https://github.com/oliverban) Can you tell me what you used to prepare the dataset and share your config for training the lore?

It's just the default hunyuan one but changed the model to be Wan I2V 14B 720p and then trained on 50 videos (with manual captions).

resolutions = [544]
enable_ar_bucket = true
min_ar = 0.5
max_ar = 2.0
num_ar_buckets = 7
frame_buckets = [ 33, 65, 81 ]

[[directory]]
path = 'PATH'
num_repeats = 10

I had 80GB GPU so the settings in the config are just default (for hunyuan) but pushed a little higher to utilize the full capacity of the VRAM.

**Comment by oliverban** [comment]
> Same problem.I don't know how to use lora trained by i2v.

@tdrussell do you have any insights? Do we need to convert it to be comfy compatible?

**Comment by Vova-B** [comment]
> > [@oliverban](https://github.com/oliverban) Can you tell me what you used to prepare the dataset and share your config for training the lore?
> 
> It's just the default hunyuan one but changed the model to be Wan I2V 14B 720p and then trained on 50 videos (with manual captions).
> 
> resolutions = [544] enable_ar_bucket = true min_ar = 0.5 max_ar = 2.0 num_ar_buckets = 7 frame_buckets = [ 33, 65, 81 ]
> 
> [[directory]] path = 'PATH' num_repeats = 10
> 
> I had 80GB GPU so the settings in the config are just default (for hunyuan) but pushed a little higher to utilize the full capacity of the VRAM.

How long did it take you to train the lore and how good is the result? I set the lora for i2v to train on two Nvidia H200 video cards and my approximation in time is approximately 16 hours. At the same time, the loss graph does not decrease but remains at approximately the same values ​​throughout the training

**Comment by tdrussell** [comment]
>Do we need to convert it to be comfy compatible?

You should not need to convert it. And the keys it is complaining about not loading are what they should be. Which is why this is odd. Have you double checked your comfyui workflow, made sure that you aren't trying to load the lora on a different model for instance? Are you able to share the workflow?

>At the same time, the loss graph does not decrease but remains at approximately the same values ​​throughout the training

This can be normal depending on your training dataset. The loss is naturally very noisy due to training process choosing random noise and timesteps for each example. Ideally you would use a held-out validation dataset, because that loss is stabilized and you could see it smoothly decrease during training. And it also would tell you when/if you start overfitting.

The default settings might not be optimal. i2v might be different from t2v. Video could need different settings from image-only training. I just don't know. The learning rate of 2e-5 is a bit on the lower side, but it works fairly well for models like flux, or image-only training on HunyuanVideo or Wan.

**Comment by oliverban** [comment]
> > Do we need to convert it to be comfy compatible?
> 
> You should not need to convert it. And the keys it is complaining about not loading are what they should be. Which is why this is odd. Have you double checked your comfyui workflow, made sure that you aren't trying to load the lora on a different model for instance? Are you able to share the workflow?

So it _**seems**_ to work when I turn of Torch compile in Native workflow.

**Comment by Vova-B** [comment]
> > > Do we need to convert it to be comfy compatible?
> > 
> > 
> > You should not need to convert it. And the keys it is complaining about not loading are what they should be. Which is why this is odd. Have you double checked your comfyui workflow, made sure that you aren't trying to load the lora on a different model for instance? Are you able to share the workflow?
> 
> So it _**seems**_ to work when I turn of Torch compile in Native workflow.

To import a trained WAN LoRA into ComfyUI, you need to copy two files into the `models/loras/` folder:
	1.	`adapter_model.safetensor`
	2.	`adapter_config.json`

After that, you can import the LoRA and load it into your model in ComfyUI without any issues.

**Comment by oliverban** [comment]
> before the torch.compile happen

Awesome! So it would be Load Diff Model > LoraModelOnly (Here we load lora) > Patch Model Patcher > TorchCompileWan > ModelSamplingSD3 > Tea/FETA > and finally Sampler? 

Thanks so much for clarification!

**Comment by kijai** [comment]
> > before the torch.compile happen
> 
> Awesome! So it would be Load Diff Model > LoraModelOnly (Here we load lora) > Patch Model Patcher > TorchCompileWan > ModelSamplingSD3 > Tea/FETA > and finally Sampler?
> 
> Thanks so much for clarification!
> 
> EDIT: I have no idea what changed but my generations have jumped from 10-15 min to over 30 min. I have no idea why, even dragging in the video I know rendered in 13 min last time now shows me 50 minutes.

A note about the (hacky like mentioned) `Patch Model Patch Order` -node people recently noticed, and I personally confirmed is that it can mess with the automatic memory management of ComfyUI, your case sounds the same. As a work around I added option to force the partial load which did solve that issue, but possibly causes it to offload too much, so might not be most efficient, however if you already are on the edge of your VRAM like most of us are with this model, it shouldn't matter much:

![Image](https://github.com/user-attachments/assets/10ea68c8-3bfb-4535-b0dd-7376ef5357d9)

This node was supposed to be a temporary workaround to the issue, I've raised the issue of torch.compile and LoRAs with the Comfy team numerous times, to comfyanon himself as well, they are aware of it. I wouldn't dare to assume I know a way to do this cleanly without affecting something else.

**Comment by oliverban** [comment]
> > > before the torch.compile happen
> > 
> > 
> > Awesome! So it would be Load Diff Model > LoraModelOnly (Here we load lora) > Patch Model Patcher > TorchCompileWan > ModelSamplingSD3 > Tea/FETA > and finally Sampler?
> > Thanks so much for clarification!
> > EDIT: I have no idea what changed but my generations have jumped from 10-15 min to over 30 min. I have no idea why, even dragging in the video I know rendered in 13 min last time now shows me 50 minutes.
> 
> A note about the (hacky like mentioned) `Patch Model Patch Order` -node people recently noticed, and I personally confirmed is that it can mess with the automatic memory management of ComfyUI, your case sounds the same. As a work around I added option to force the partial load which did solve that issue, but possibly causes it to offload too much, so might not be most efficient, however if you already are on the edge of your VRAM like most of us are with this model, it shouldn't matter much:
> 
> ![Image](https://github.com/user-attachments/assets/10ea68c8-3bfb-4535-b0dd-7376ef5357d9)
> 
> This node was supposed to be a temporary workaround to the issue, I've raised the issue of torch.compile and LoRAs with the Comfy team numerous times, to comfyanon himself as well, they are aware of it. I wouldn't dare to assume I know a way to do this cleanly without affecting something else.

I actually am using it on a 80GB card so VRAM is not an issue and I am trying to not offload anything at all actually since that would mean fastest inference/iteration time, right? I did try with the new one and setting it to disabled.

**Comment by kijai** [comment]
> I actually am using it on a 80GB card so VRAM is not an issue and I am trying to not offload anything at all actually since that >would mean fastest inference/iteration time, right? I did try with the new one and setting it to disabled. But I might also be asking >just too much maybe? <3

If you mean never even offloading the models, as in running with something like `--high-vram` or `--gpu-only`, that wouldn't usually have huge impact on the overall runtime, at least locally where moving the model from between GPU and CPU takes only few seconds, no effect on inference time itself.

Filling up VRAM and activating the automated layer offloading in ComfyUI however would slow down the inference a lot, it should not happen on 80GB unless you really crank up the settings **and** don't offload the text encoder/VAE/clip_vision models.

13mins to 50mins sounds like slowdown that could only be caused by that to me.

**Comment by kijai** [comment]
>  does the WAN Enhance-a-video (native) slow inference down?

Technically it does, but very very slightly, usually I find it worth it for T2V at least. Not really sure if it's good for I2V yet.

**Comment by oliverban** [comment]
> > does the WAN Enhance-a-video (native) slow inference down?
> 
> Technically it does, but very very slightly, usually I find it worth it for T2V at least. Not really sure if it's good for I2V yet.

I see! Yeah, Because that one seems to also add at least 10-15 minutes for me when active with 2.0 in the value. I'll play around, thanks again! <3

**Comment by kijai** [comment]
> I see! Yeah, Because that one seems to also add at least 10-15 minutes for me when active with 2.0 in the value. I'll play around, thanks again! <3

That really shouldn't happen, and I've never seen it add more than milliseconds per iteration... suppose it can mess with the compile, whole torch.compile usage in native workflows is just messy right now, often when changing settings you need to reload the model from start to actually apply them when using it.

**Comment by oliverban** [comment]
> > I see! Yeah, Because that one seems to also add at least 10-15 minutes for me when active with 2.0 in the value. I'll play around, thanks again! <3
> 
> That really shouldn't happen, and I've never seen it add more than milliseconds per iteration... suppose it can mess with the compile, whole torch.compile usage in native workflows is just messy right now, often when changing settings you need to reload the model from start to actually apply them when using it.

Great behind the scenes here! I'm sure the torch kinks will be ironed out eventually, it's too valuable to ignore imho!

**Comment by mdkberry** [comment]
any luck with this? I trained a Wan Lora using t2v 1.3B model, and whenever I use it with Wan i2v models I get endless `ERROR lora diffusion_model.blocks.12.self_attn.k.weight shape '[5120, 5120]' is invalid for input of size 23592` type messages. it finishes and works, but is considerably slowed down. I also dont use torch node or Kijais workflow only his patch sage attn node, but do have teacache, sage attn nodes.
